core/dbt/include/global_project/macros/materializations/models/incremental/merge.sql

```diff
@@ -1,9 +1,8 @@
-{% macro get_merge_sql(target, source, unique_key, dest_columns, predicates=none) -%}
+{% macro get_merge_sql(target, source, unique_key, dest_columns, predicates=[]) -%}
```

Comment:
> I've changed the default value of predicates to be an empty list from none, that may have been done for a reason I'm unaware of though!

I think because this makes it a [mutable default argument](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments), which is [the source of all evil](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/)

Reply:
Reverted this, but I have left the `get` [here](https://github.com/dbt-labs/dbt-core/pull/5679/files#diff-166e98f1f0a0a6bb94f26740cb121dd7eec6c5ebde8ba9a658bc7fe86e3fa881R53) to default return an empty list vs none. I _think_ that's ok.